lib/serviceClient.ts

```diff
@@ -187,7 +192,10 @@ export class ServiceClient {
 
     let serviceClientCredentials: ServiceClientCredentials | undefined;
     if (isTokenCredential(credentials)) {
-      serviceClientCredentials = new AzureIdentityCredentialAdapter(credentials);
+      serviceClientCredentials = new AzureIdentityCredentialAdapter(
+        credentials,
+        options?.baseUri ? `${options.baseUri}/.default` : undefined
```

Comment:
Should we also set `this.baseUri` to `options.baseUri` if it exists? It's a protected member on this class, but I don't feel too strongly about it.

Reply:
Thanks! I thought about this, then I forgot to do it 🤦‍♂️